contribute-to-tidb/review-a-pr.md

```diff
@@ -7,46 +7,163 @@ This guide is for all contributors that want to help with reviewing code contrib
 
-## 1. Is the Contribution Well-Described?
+## 1. Things to know before the code review
 
-Check whether the contribution is sufficiently well-described to support a good review. Trivial changes and fixes do not need a long description. If the implementation is exactly according to a prior discussion on an issue or the [TiDB Internals forum](https://internals.tidb.io/), only a short reference to that discussion is needed. If the implementation is different from the agreed approach in the consensus discussion, a detailed description of the implementation is required for any further review of the contribution.
+Why is code review important?
 
-Any pull request that changes functionality or behavior needs to describe the big picture of these changes, so that reviews know what to look for \(and don't have to dig through the code to hopefully understand what the change does\).
+* Erase bugs early
+* Make the project maintainable in the long run
 
-A contribution is well-described if the following questions 2, 3, and 4 can be answered without looking at the code.
+Some high level guidelines about code review in PingCAP (and mostly applies the TiDB community also):
 
-## 2. Is There Consensus that the Change or Feature Should Go into TiDB?
+### Code review is more important than writing your own PR
 
-This question can be directly answered from the linked GitHub issue. For pull requests that are created without prior consensus, a discussion in issue or the [TiDB Internals forum](https://internals.tidb.io/) to seek consensus will be needed.
+Modern software development is about cooperation, and nowadays software is no longer one single man’s piece of art. To speed up the development and make the processing scale, it is important to avoid blocking. If the members in an organization give higher priority for writing their own code, others waiting for code review get blocked, and there will be a lot of pending PRs for review. Eventually most of the developers get blocked, the whole team slows down.
 
-For trivial pull requests, consensus needs to be checked in the pull request.
+### It's the author's responsibility to get the PR merged
 
-## 3. Does the Contribution Need Attention from some Specific Members?
+We had a fierce debate on how to speed up the review process long long ago. What the role the reviewer and PR owner should play and what is each of their responsibility to get the PR merged faster.
+After the meeting, most of us agree with the conclusion that the PR owner is responsible for getting his PR merged. He is the one who most wanted to merge the PR into the repository. He is willing to lead that change. So he should try to do his best to have someone review his PR. The code reviewers should review the PR in time and give feedback, but it is not their responsibility to push the PR to be merged.
```

Comment:
We can rephrase this section in "I" form instead of "he" for warming reading experience and avoid risk of sexism.

```suggestion
We had a fierce debate on how to speed up the review process long long ago. What the role the reviewer and PR owner should play and what is each of their responsibility to get the PR merged faster.

After the meeting, most of the participants agree with that the PR owner is responsible for getting PR merged.

If I am the PR author, I am the one who wants to merge the PR into the repository most. I'm willing to lead that change. So I'm supposed to do my best to have someone review the PR. The reviewers should review the PR in time and give feedback, but it is not their responsibility to push the PR to be merged.
```